### DISCUSSION TITLE
[SDK] Whether the Wagmi + Ethers combination can be used normally ?

### DISCUSSION THREAD

**Discussion by dxiongya** [zkSync-Community-Hub/zksync-developers#430] [SDKs] [answered]
### Environment

Mainnet

### Acknowledgement

- [X] I understand that issues are expected with the Go SDK as they are being updated.

### Issue Description

HI, I am a new developer who has just joined Zksync era and I have some questions about dapp.

I see that the `zksync-ethers` library extends ethers, but I'm not sure if I can still use it in the Wagmi + Ethers combination in my React project? Is there any behavior that doesn't work properly if it involves a contract invocation? need to use zksync-ethers instead?

I have seen the official tutorial, Wagmi + VIEM, but did not see the official introduction of this in the dapp demo? Or did I miss it?

I also saw in the document that zksync-ethers is added functionality to ethers. Is this kind of additional functionality important in the formal dapp development? If so, will it be a problem when using viem and ethers in conjunction with wagmi?

### Expected Behavior

none

### Code Example

none

**Comment by dxiongya** [comment]
I think I saw something, can I understand that?
Zksync-ethers simply adds special treatment to deploying smart contracts or using unique zkSync features, like account abstraction, without adjusting the other internal logic of ethers?
So ethers + wagmi / view + wagmi works, unless we use special features like zksync and we need zksync-ethers to intervene, right?

**Reply by bxpana** [reply, marked as answer]
Yes, you're correct in your understanding. The `zksync-ethers` library extends the functionality of ethers to support unique zkSync features like account abstraction and deploying smart contracts. If your dApp doesn't require these specific zkSync features, you can use ethers in combination with Wagmi or Vue without any issues.
However, if you want to deploy smart contracts or enable advanced zkSync features, like account abstraction, then you need to use the `zksync-ethers` library. This is because these features require providing additional fields to those that Ethereum transactions have by default.

**Reply by dxiongya** [reply]
thx！What about viem?

**Reply by dutterbutter** [reply]
@chongqiangchen viem supports zkSync: https://viem.sh/zksync - we are actively working to add additional zkSync support to viem and will have parity with zksync-ethers in the future.
